### PR TITLE
249140: Agent Pool rename and properties changes

### PR DESCRIPTION
**Pull request by ajaysoni-cap**
# **What this PR does / why we need it**:
Correct the cluster and agent pool properties to support the version upgrades.
Update to the agent pool names as per the latest design.
Link to the relevant work items: [AB#249140](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/249140)

# **Special notes for your reviewer**
None

# Testing
n/a

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: infra/core/managed-cluster/aks-cluster.bicep [hunk 1]
```diff
@@ -60,6 +60,36 @@ var asoPlatformTeamMiRbacs = [
   }
 ]
 
+var systemNodePool = {
+  name: 'npsystem01'
+  mode: 'System'
+  count: cluster.npSystem.count
+  vmSize: 'Standard_DS2_v2'
```
Comment (DaRoszko) [inline infra/core/managed-cluster/aks-cluster.bicep:67]: DS3_V2 - but also needs to be variables that can change on environments.

**Review by DaRoszko — APPROVED** [review]

**Review by kbabigumira — APPROVED** [review]

File: infra/core/managed-cluster/aks-cluster.bicep [hunk 2]
```diff
@@ -60,6 +60,36 @@ var asoPlatformTeamMiRbacs = [
...
+  osDiskSizeGB: cluster.npSystem.osDiskSizeGB
+  osDiskType: 'Ephemeral'
+  osType: 'Linux'
+  osSKU: 'Ubuntu'
+  minCount: cluster.npSystem.minCount
+  maxCount: cluster.npSystem.maxCount
+  vnetSubnetId: resourceId(vnet.resourceGroup, 'Microsoft.Network/virtualNetworks/subnets', vnet.name, vnet.subnetClusterNodes)
+  enableAutoScaling: true
+  enableCustomCATrust: false
```
Comment (DaRoszko) [inline infra/core/managed-cluster/aks-cluster.bicep:76]: custom CA trust on in future?

File: infra/core/managed-cluster/aks-cluster.bicep [hunk 3]
```diff
@@ -60,6 +60,36 @@ var asoPlatformTeamMiRbacs = [
...
+  enableFIPS: false
+  enableEncryptionAtHost: false
```
Comment (DaRoszko) [inline infra/core/managed-cluster/aks-cluster.bicep:78]: prob should be true for encryption at host...

File: infra/core/managed-cluster/aks-cluster.bicep (hunk 1 above)
Comment (DaRoszko) [reply infra/core/managed-cluster/aks-cluster.bicep:67]: 2 diff vars for 2 diff nodepools, user, system, blah, should match naming conventions

File: infra/core/managed-cluster/aks-cluster.bicep (hunk 3 above)
Comment (ajaysoni-cap) [reply infra/core/managed-cluster/aks-cluster.bicep:78]: Will pick it up in the next pr.

File: infra/core/managed-cluster/aks-cluster.bicep (hunk 2 above)
Comment (ajaysoni-cap) [reply infra/core/managed-cluster/aks-cluster.bicep:76]: Will pick it up in the next pr.

**Pull request merged**